### PR TITLE
Speed up test worker startup

### PR DESCRIPTION
**Pull request by oehme**
Our test processes are very fast compared to other build tools, once they are warmed up. But in the "100 projects with only 100 tests each" scenario, we were ~10% slower than Maven, because our test processes take a bit longer to start up because of that added complexity.

These changes mitigate that by:

- doing less classloading (especially Guava)
- not initializing the native services for test workers
- starting the worker health check only after the first delay
- making some logging services cheaper to create

This closes the performance gap for the scenario above and should also slightly improve the Gradle client startup time.

### CODE REVIEW

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/experiment/maven/JavaTestGradleVsMavenPerformanceTest.groovy [hunk 1]
```diff
@@ -60,17 +60,15 @@ class JavaTestGradleVsMavenPerformanceTest extends AbstractGradleVsMavenPerforma
         def results = runner.run()
 
         then:
-        if (gradleTask != "test" || testProject != MEDIUM_JAVA_MULTI_PROJECT) { //there seems to be some inefficiency when testing with --parallel
-            results.assertFasterThanMaven()
-        }
+        results.assertFasterThanMaven()
```
Comment (melix) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/experiment/maven/JavaTestGradleVsMavenPerformanceTest.groovy]: :+1:

File: subprojects/logging/src/main/java/org/gradle/internal/logging/console/StyledTextOutputBackedRenderer.java [hunk 2]
```diff
@@ -44,6 +44,9 @@ public void onOutput(OutputEvent event) {
         if (event instanceof LogLevelChangeEvent) {
             LogLevelChangeEvent changeEvent = (LogLevelChangeEvent) event;
             debugOutput = changeEvent.getNewLogLevel() == LogLevel.DEBUG;
+            if (debugOutput && dateFormat == null) {
+                dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
```
Comment (melix) [inline subprojects/logging/src/main/java/org/gradle/internal/logging/console/StyledTextOutputBackedRenderer.java]: :x: Can this be called concurrently? If so, there's a race condition here. A thread can see `dateFormat` not null, while initializing `SimpleDateFormat` is not complete.

File: subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java [hunk 3]
```diff
@@ -117,6 +127,9 @@ public void restore(Snapshot state) {
     }
 
     public ColorMap getColourMap() {
+        if (colourMap == null) {
+            colourMap = new DefaultColorMap();
```
Comment (melix) [inline subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java]: :x: This is not thread safe. One thread can read `colourMap` not null, before the other thread finished initializing `DefaultColorMap`. I reckon this is called from `attachAnsiConsole`, but the method is public. So to be safe, it needs to acquire the lock.

File: subprojects/logging/src/main/java/org/gradle/internal/logging/console/StyledTextOutputBackedRenderer.java (hunk 2 above)
Comment (oehme) [inline subprojects/logging/src/main/java/org/gradle/internal/logging/console/StyledTextOutputBackedRenderer.java]: No, everything below `OutputEventRenderer` does not have to be safe.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleVsMavenPerformanceTestRunner.groovy [hunk 4]
```diff
@@ -69,7 +69,6 @@ class GradleVsMavenPerformanceTestRunner extends AbstractGradleBuildPerformanceT
             invocationCount = runs
             projectName(testProject).displayName("Maven $commonBaseDisplayName").invocation {
                 tasksToRun(equivalentMavenTasks.split(' ')).mavenOpts(jvmOpts.collect {it.toString()}).args(mvnArgs.collect {it.toString()})
-                    .args('-q', '-Dsurefire.printSummary=false')
```
Comment (oehme) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleVsMavenPerformanceTestRunner.groovy]: FTR this didn't have any effect on the test results. Spooling the output to a file is blazing fast.

File: subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java (hunk 3 above)
Comment (oehme) [inline subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java]: Fixed

**Pull request merged**